### DISCUSSION TITLE
application frozen with cx_freeze stack trace refers to python directory

### DISCUSSION THREAD

**Discussion by jschwar313** [marcelotduarte/cx_Freeze#1612] [Q&A] [answered]
I froze an application built in python 3.10.7 with cx_freeze version 6.11.1 and it refers back to my python directory, not the installation directory of my application.  Is that correct?  I thought that python would be put in the installation directory and it would be used there.  Am I wrong?

**Comment by marcelotduarte** [comment, marked as answer]
> I froze an application built in python 3.10.7 with cx_freeze version 6.11.1 and it refers back to my python directory, not the installation directory of my application. Is that correct? 

Yes, it is.

> I thought that python would be put in the installation directory and it would be used there. Am I wrong?

You can change this behavior, using the [replace_paths](https://cx-freeze.readthedocs.io/en/latest/setup_script.html?cmdoption-arg-replace_paths#cmdoption-arg-replace_paths) option/argument.

One sample use the simplest form of it: https://github.com/marcelotduarte/cx_Freeze/blob/main/samples/sqlite/setup.py

**Comment by jschwar313** [comment]
What would happen if I put this application on another computer without python installed in that directory?  Would it use my application directory then?  Would the trace back stack trace display the replaced path then?

**Reply by marcelotduarte** [reply]
No, No, yes.

The path is only for reference because the original file is not distributed, only a compiled file. For instance, a module file test.py is distributed as a test.pyc but the reference it to the source to help debug.

**Comment by jschwar313** [comment]
What does  "replace_paths": [("*", "")], mean?

**Reply by marcelotduarte** [reply]
The answer is on the link https://cx-freeze.readthedocs.io/en/latest/setup_script.html?cmdoption-arg-replace_paths#cmdoption-arg-replace_paths
